### PR TITLE
[IMP] delivery: Make delivery line creation and selection hookable

### PR DESCRIPTION
**Pull request by mt-software-de**
This PR does not change the current behavior, it makes the creation of a sale.order.line is_delivery hookable. 

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by vava-odoo — COMMENTED** [review]
Hi @mt-software-de 
Thanks for the work.
I'm not against such a change but, out of curiosity, could you tell us more about the use-case behind these changes?
Cheers,

**Comment by mt-software-de** [comment]
> Hi @mt-software-de Thanks for the work. I'm not against such a change but, out of curiosity, could you tell us more about the use-case behind these changes? Cheers,

One example would be that we want to track want to track what created/updated the delivery_line, if it is created from the sale.order or from the stock.picking.

I wanted to keep the diff as small as possible, that's why i didn't changed the line length. But if this is fine, then i will clean the code.

File: addons/delivery/models/sale_order.py [hunk 1]
```diff
@@ -91,12 +91,9 @@ def action_open_delivery_wizard(self):
             }
         }
 
-    def _create_delivery_line(self, carrier, price_unit):
-        SaleOrderLine = self.env['sale.order.line']
-        context = {}
+    def _prepare_delivery_line_vals(self, carrier, price_unit):
         if self.partner_id:
             # set delivery detail in the customer language
-            context['lang'] = self.partner_id.lang
```
Comment (vava-odoo) [inline addons/delivery/models/sale_order.py:99]: I forgot, put it back please.

**Review by vava-odoo — APPROVED** [review]
Thanks for the changes. It is fine by me! (just a small comment I forgot).

> One example would be that we want to track want to track what created/updated the delivery_line, if it is created from the sale.order or from the stock.picking.

Thanks for sharing. Makes sense, indeed.
> I wanted to keep the diff as small as possible, that's why i didn't changed the line length. But if this is fine, then i will clean the code.

Well, it is the latest stable, so it is worth cleaning a bit (but not everyone agrees on that).

File: addons/delivery/models/sale_order.py (hunk 1 above)
Comment (vava-odoo) [reply addons/delivery/models/sale_order.py:99]: From what I know of it, it is *implicitly* used by the translation engine (`_(...)`) below.

Comment (mt-software-de) [reply addons/delivery/models/sale_order.py:99]: ok thanks. You are right.

**Review by Feyensv — APPROVED** [review]
@robodoo r+
Thanks for the changes :+1:

**Pull request closed without merging**